### PR TITLE
Fix for config variant stock 736

### PR DESCRIPTION
**Pull request by minesh93**
### Description
This will show product variation if the configuration value of `cataloginventory/options/show_out_of_stock` is set to 1.

Out of stock items are then filtered out via `Magento\CatalogInventory\Helper\Stock::addInStockFilterToCollection`

### Fixed Issues
1. #736 Configurable products with "out of stock" return as product: null

### Manual testing scenarios (*)
Create a configurable product with one of the variants being out of stock. Test with both `cataloginventory/options/show_out_of_stock` being enabled and disabled.

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/graphql-ce?pullRequest=747) <br/>All committers have signed the CLA.

File: app/code/Magento/ConfigurableProductGraphQl/Model/Variant/Collection.php [hunk 1]
```diff
@@ -41,6 +43,16 @@ class Collection
...
 
+    /**
+     * @var Stock
+     */
+    private $stockHelper;
+
+    /**
+     * @var StockConfigurationInterface
+     */
+    protected $stockConfig;
```
Comment (naydav) [inline app/code/Magento/ConfigurableProductGraphQl/Model/Variant/Collection.php]: Should be private

File: app/code/Magento/ConfigurableProductGraphQl/Model/Variant/Collection.php [hunk 2]
```diff
@@ -143,13 +161,19 @@ private function fetch() : array
...
+        $showOutOfStock = $this->stockConfig->isShowOutOfStock();
+
         foreach ($this->parentProducts as $product) {
             $attributeData = $this->getAttributesCodes($product);
             /** @var ChildCollection $childCollection */
             $childCollection = $this->childCollectionFactory->create();
             $childCollection->setProductFilter($product);
             $childCollection->addAttributeToSelect($attributeData);
 
+            if (!$showOutOfStock) {
```
Comment (naydav) [inline app/code/Magento/ConfigurableProductGraphQl/Model/Variant/Collection.php]: @minesh93, please cover this functionality with API-functional tests
thanks

Examples:
https://github.com/magento/graphql-ce/tree/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Customer
https://github.com/magento/graphql-ce/tree/2.3-develop/dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/Customer

Guide:
https://devdocs.magento.com/guides/v2.3/graphql/functional-testing.html

**Review by naydav — CHANGES_REQUESTED** [review]

**Review by naydav — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @naydav, thank you for the review.
[ENGCOM-5372](https://jira.corp.magento.com/browse/ENGCOM-5372) has been created to process this Pull Request
:eight_spoked_asterisk: @naydav, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by magento-engcom-team** [comment]
@minesh93 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Pull request merged**